enhancements/storage/csi-vsphere-operator.md

```diff
@@ -152,6 +152,64 @@ If we detect an existing driver present in the cluster then cluster will be mark
 
 We will ensure that these steps are documented in 4.10 docs.
 
+#### Configure CSI driver topology
+
+A customer may configure a topology for the CSI driver by using following fields in `ClusterCSIDriver` object by specifying
+field `.driverConfig.vsphere.tagCategories`
+
+
+```go
+// ClusterCSIDriverSpec is the desired behavior of CSI driver operator
+type ClusterCSIDriverSpec struct {
+    ...
+    ...
+    // `driverConfig` field can be used to optionally specify
+    // driver specific configuration.
+    // If this field is nil, platform chosen default configuration
+    // will be applied.
+    DriverConfig *CSIDriverConfigSpec `json:"driverConfig,omitempty"`
+}
+
+// CSIDriverConfigSpec defines configuration spec that can be
+// used to optionally configure a specific CSI Driver.
+// +union
+type CSIDriverConfigSpec struct {
+    // `driverName` indicates which of the driver we are
+    // installing
+    // +unionDiscriminator
+    // +required
+    DriverName CSIDriverName `json:"driverName"`
+
+    // `vsphere` field is used to configure vsphere CSI driver
+    // specific properties
+    VSphere VSphereCSIDriverConfigSpec `json:"vsphere"`
```

Comment:
`json: vSphere`?

Reply:
I was thinking if someone is going to edit the json by hand, it will look like:

```json
{
"spec": {
  "driverConfig": {
     "vsphere": {
         "tagCategories": ["foo", "bar"]
      }
  }
}
```

Basically we don't have to be so verbose in user editable json. But I expect another round of back and forth during api reviews, so we can have these discussions there too.

Reply:
So basically because outer key already has word "driverConfig" in it - I dropped `CSIDriverConfig` suffix from the key, because it is redundant.